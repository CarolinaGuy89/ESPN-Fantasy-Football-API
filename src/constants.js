/**
 * Maps `slotCategoryId`'s numerical enum to readable positions.
 * @type {Object}
 */
const slotCategoryIdToPositionMap = {
  0: 'QB',
  1: 'TQB',
  2: 'RB',
  3: 'RB/WR',
  4: 'WR',
  5: 'WR/TE',
  6: 'TE',
  7: 'OP',
  8: 'DT',
  9: 'DE',
  10: 'LB',
  11: 'DL',
  12: 'CB',
  13: 'S',
  14: 'DB',
  15: 'DP',
  16: 'D/ST',
  17: 'K',
  18: 'P',
  19: 'HC',
  20: 'Bench',
  21: 'IR',
  22: 'Unknown?', // TODO: Figure out what this is
  23: 'RB/WR/TE',
  24: 'Unknown?' // TODO: Figure out what this is
};

export { slotCategoryIdToPositionMap };
