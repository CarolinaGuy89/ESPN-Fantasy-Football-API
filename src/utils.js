import _ from 'lodash';

const setWithWarning = (objValue, newValue, key, object) => {
  if (object[key]) {
    console.warn(`espn-fantasy-football-api: Duplicate key ${key} in flatten response!`);
  }

  _.set(object, key, newValue);
};

const flattenObject = (object) => {
  const flatObject = {};

  _.forEach(object, (value, key) => {
    if (_.isPlainObject(value)) {
      _.assignWith(flatObject, flattenObject(value), setWithWarning);
    } else {
      setWithWarning(flatObject, key, value);
    }
  });

  return flatObject;
};

const flattenObjectSansNumericKeys = (object) => {
  const flatObject = {};

  _.forEach(object, (value, key) => {
    if (_.isPlainObject(value) && !_.some(_.keys(value), (k) => !_.isNaN(Number(k)))) {
      _.assignWith(flatObject, flattenObjectSansNumericKeys(value), setWithWarning);
    } else {
      setWithWarning(flatObject, key, value);
    }
  });

  return flatObject;
};

export {
  flattenObject,
  flattenObjectSansNumericKeys
};
