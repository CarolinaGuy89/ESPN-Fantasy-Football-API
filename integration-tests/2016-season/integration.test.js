import 'dotenv/config'; // Create a root .env file defining LEAGUE_ID, ESPN_S2, and SWID

import { Client } from '../../src/index';

jest.setTimeout(10000);

describe('2016 season integration tests', () => {
  let client;
  let leagueId;
  let seasonId;
  let scoringPeriodId;

  beforeEach(() => {
    leagueId = process.env.LEAGUE_ID;
    seasonId = 2016;
    scoringPeriodId = 1;

    client = new Client({
      leagueId,
      espnS2: process.env.ESPN_S2,
      SWID: process.env.SWID
    });
  });

  describe('getHistoricalScoreboardForWeek', () => {
    test('returns a populated array of Boxscores', async () => {
      const scoreboards = await client.getHistoricalScoreboardForWeek({
        seasonId, matchupPeriodId: scoringPeriodId, scoringPeriodId
      });

      expect(scoreboards).toMatchSnapshot();
    });
  });
});
