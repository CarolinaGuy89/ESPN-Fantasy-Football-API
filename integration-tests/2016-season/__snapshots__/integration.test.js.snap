// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`2016 season integration tests getHistoricalScoreboardForWeek returns a populated array of Boxscores 1`] = `
[
  Boxscore {
    "awayRoster": [],
    "awayScore": 106.06,
    "awayTeamId": 4,
    "homeRoster": [],
    "homeScore": 96.56,
    "homeTeamId": 1,
  },
  Boxscore {
    "awayRoster": [],
    "awayScore": 87.86,
    "awayTeamId": 2,
    "homeRoster": [],
    "homeScore": 86.32,
    "homeTeamId": 9,
  },
  Boxscore {
    "awayRoster": [],
    "awayScore": 123.3,
    "awayTeamId": 5,
    "homeRoster": [],
    "homeScore": 103.76,
    "homeTeamId": 3,
  },
  Boxscore {
    "awayRoster": [],
    "awayScore": 94.72,
    "awayTeamId": 8,
    "homeRoster": [],
    "homeScore": 109.6,
    "homeTeamId": 12,
  },
]
`;
